### ISSUE TITLE
Add `maxNesting` option to `object-curly-newline`.

### ISSUE THREAD

**Issue by nkt** [eslint/eslint#6549]
Configuration:
```json
{
  "rules": {
    "object-curly-newline": [2, { 
      "multiline": true 
    }]
  }
}
```

Following code consider error:

```js
// test.js

const defaultConfig = {
  endpoints: {
    http: null
  }
};

```

Result:
```
❯ eslint test.js 

test.js
  4:14  error  Unexpected a line break after this open brace    object-curly-newline
  6:3   error  Unexpected a line break before this close brace  object-curly-newline

✖ 2 problems (2 errors, 0 warnings)
```

I suggest add `maxNesting` option, which allows using linebreaks if object riches `maxNesting`.
 
- `{ foo: { bar: {} } }` - nesting = 2
- `{ foo: { bar: { baz: {} } } }` - nesting = 3

**Comment by eslintbot** [comment]
Hi @nkt, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by nkt** [comment]
I'm using latest ESLint (v2.13.1).

**Comment by vitorbal** [comment]
Hi @nkt, could you provide us with the actual output from ESLint after running it through your example code? Could you also explain a bit more how the new option you're proposing would look like? It would be helpful to list out some example cases that would be considered valid and some that would be considered invalid with the new option.
Thanks!

**Comment by platinumazure** [comment]
Ping @nkt, we need more information before we can determine what sort of issue this is. Can you please answer @vitorbal's question as well as @eslintbot's questions? Thanks!

**Comment by vitorbal** [comment]
Sorry to bother you again, @nkt, could you come back to us on the questions above? Otherwise I'm afraid we don't have enough information to consider your proposal. Thanks!

**Comment by nkt** [comment]
Sorry for so long delay. I've updated first message @platinumazure @vitorbal

**Comment by BYK** [comment]
I'm 👍 to this suggestion.

**Comment by kaicataldo** [comment]
I'm closing this issues because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
